### DISCUSSION TITLE
Verifying copy_object using etag via stat_object

### DISCUSSION THREAD

**Discussion by jtan346** [minio/minio#14645] [Q&A]
Good day minio team and fellow devs,

I am using minio.py and I am looking for best practices to carry out object migrations within the same bucket.

My current method is:
1. `copy_object`
2. Verify that both old and new (copied) object has the same etags via `stat_object`
3. delete original objects

However, during verification phase I realised that my original object etag is "00000000000000000000000000000000-1" while my new object etag is valid "495219f126486f598391f353f4c895b4". Visually, both objects exist in minio UI and have the same file size, correct date modified, etc.

Hence I have 2 questions:
1. Do we need to verify copy_object commands or is it guaranteed that an object copied will be valid (non-corrupt)?
2. If so, what is the best practice to verify the copied file integrity?

Thanks!

**Comment by harshavardhana** [comment]
You are starting with a backend with existing content so of course you have no Etag to begin with.

**Reply by jtan346** [reply]
Hi! Thanks for the answer, certain items do have etag values, but not all of them, hence my confusion. Moreover, is etag the best practice when it comes to validating copied files, or is there another better way? Thanks!

**Reply by jtan346** [reply]
I have read that this is because the files are uploaded via fs commands. The target files have normal etag values. Is there a way to trigger recomputation of etag on server side?
